auslib/test/web/test_client.py

```diff
@@ -461,6 +539,35 @@ def testDeprecatedEsrVersionStyleGetsUpdates(self):
 """)
         self.assertEqual(returned.toxml(), expected.toxml())
 
+    def testGetWithResponseProducts(self):
+        ret = self.client.get('/update/4/gmp/1.0/1/p/l/a/a/a/a/1/update.xml')
+        self.assertEqual(ret.status_code, 200)
+        self.assertEqual(ret.mimetype, 'text/xml')
+        returned = minidom.parseString(ret.data)
+        expected = minidom.parseString("""<?xml version="1.0"?>
+<updates>
+    <update type="minor" version="None" extensionVersion="2.5" buildID="25">
+        <patch type="complete" URL="http://a.com/public" hashFunction="sha512" hashValue="23" size="22"/>
+        <patch type="complete" URL="http://a.com/b" hashFunction="sha512" hashValue="23" size="27777777"/>
+    </update>
+</updates>
+""")
+        self.assertEqual(returned.toxml(), expected.toxml())
+
+    def testGetWithResponseProductsWithAbsentRule(self):
```

Comment:
Great work on these tests! I'm happy to see both one and multiproduct responses being tested here.